### PR TITLE
Reorganize platformer example for readability

### PR DESCRIPTION
**Pull request by Koopa1018**
The components and systems in the platformer example are confusingly sorted: components are stored in a separate file from the systems that use them, and within those files the components & systems are jumbled chaotically with little rhyme or reason. In a crate example this complex, this organization scheme makes understanding the code a real challenge (which is a major reason I couldn't update this example in #294).

This PR reorganizes the platformer example into modules sorted roughly by "part of the game" or "game system" (camera system, climbing, inventory, etc.). Each module contains both the components and ECS systems specific to that part, to present a complete picture of what the game system does and how it works.

No game logic has been changed. This PR focuses purely on code quality.

### CODE REVIEW

File: examples/platformer/main.rs [hunk 1]
```diff
@@ -36,25 +44,17 @@ fn main() {
...
+        .add_systems(Update, walls::spawn_wall_collision)
+        .add_plugins(physics::PhysicsPlugin)
+        .add_plugins(climbing::ClimbingPlugin)
+        .add_plugins(player::PlayerPlugin)
+        .add_plugins(enemy::EnemyPlugin)
+        .add_systems(Update, inventory::dbg_print_inventory)
+        .add_systems(Update, camera::camera_fit_inside_current_level)
+        .add_plugins(misc_objects::MiscObjectsPlugin)
+        .register_ldtk_int_cell::<walls::WallBundle>(1)
+        .register_ldtk_int_cell::<climbing::LadderBundle>(2)
```
Comment (Trouv) [inline examples/platformer/main.rs:57]: Could the `LadderBundle` be registered in the `LadderPlugin`?

**Review by Trouv — COMMENTED** [review]
Great PR, much appreciated.

This really highlights how bloated and unfocused this example has become over the years. It's definitely overdue for some refactoring and this is great start.

I just have one non-NIT about the ladder registration, assuming CI passes

**Comment by Trouv** [comment]
Clippy job should be fixed after #331

**Comment by Koopa1018** [comment]
Alright, changes complete.

**Pull request merged**